src/scanner.c

```diff
@@ -2835,44 +2838,42 @@ void FormatOutput(void (*put_a_char)(Char c), const Char *format, Int arg1, Int
 
     /* '%c' print a character                                      */
     else if ( *p == 'c' ) {
-      put_a_char( (Char)arg1 );
+      put_a_char(state, (Char)arg1);
       arg1 = arg2;
     }
 
     /* '%%' print a '%' character                                  */
     else if ( *p == '%' ) {
-      put_a_char( '%' );
+      put_a_char(state, '%');
     }
 
     /* '%>' increment the indentation level                        */
     else if ( *p == '>' ) {
-      put_a_char( '\01' );
+      put_a_char(state, '\01');
       while ( --prec > 0 )
-        put_a_char( '\01' );
+        put_a_char(state, '\01');
     }
 
     /* '%<' decrement the indentation level                        */
     else if ( *p == '<' ) {
-      put_a_char( '\02' );
+      put_a_char(state, '\02');
       while ( --prec > 0 )
-        put_a_char( '\02' );
+        put_a_char(state, '\02');
     }
 
     /* else raise an error                                         */
     else {
       for ( p = "%format error"; *p != '\0'; p++ )
-        put_a_char( *p );
+        put_a_char(state, *p);
     }
 
   }
 
 }
 
 
-/* TL: static KOutputStream TheStream; */
-
-static void putToTheStream( Char c) {
-  PutChrTo(STATE(TheStream), c);
+static void putToTheStream(void *state, Char c) {
+  PutChrTo((KOutputStream)state, c);
```

Comment:
As you can see, here we pass in a ˋKOutputStreamˋ

Reply:
fwiw I have stabbed and prodded at getting rid of this contraption, but its not easy.

 My "vison" os for my PRs #1815 et al to make this go away in the end.

Reply:
Actually, I don't think it's that bad after this PR; it's close to gow a generic C stream API would look: an opaque pointer to the "stream", which is passedto functions acting on it.

Anyway: babysteps